pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint/topologyspreadconstraint.go

```diff
@@ -47,27 +48,65 @@ type topology struct {
 	pods []*v1.Pod
 }
 
-// nolint: gocyclo
-func RemovePodsViolatingTopologySpreadConstraint(
-	ctx context.Context,
-	client clientset.Interface,
-	strategy api.DeschedulerStrategy,
-	nodes []*v1.Node,
-	podEvictor *evictions.PodEvictor,
-	evictorFilter *evictions.EvictorFilter,
-	getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc,
-) {
-	strategyParams, err := validation.ValidateAndParseStrategyParams(ctx, client, strategy.Params)
+// RemovePodsViolatingTopologySpreadConstraint evicts pods which violate their topology spread constraints
+type RemovePodsViolatingTopologySpreadConstraint struct {
+	handle         framework.Handle
+	args           *componentconfig.RemovePodsViolatingTopologySpreadConstraintArgs
+	taintFilterFnc func(taint *v1.Taint) bool
+	evictorFilter  *evictions.EvictorFilter
+	podFilter      podutil.FilterFunc
+}
+
+var _ framework.DeschedulePlugin = &RemovePodsViolatingTopologySpreadConstraint{}
```

Comment:
I believe this was intended to be a `BalancePlugin` as opposed to `DeschedulePlugin`

See a reference to the [proposal](https://github.com/kubernetes-sigs/descheduler/issues/837#issuecomment-1148878383)

Deschedule Plugin does not takes other pods into account, which we know that topology spread constraint selector does.